stdlib/src/utils/span.mojo

```diff
@@ -180,9 +180,18 @@ struct Span[
         var end: Int
         var step: Int
         start, end, step = slc.indices(len(self))
-        debug_assert(
-            step == 1, "Slice must be within bounds and step must be 1"
-        )
+
+        if step < 0:
+            step = -step
+            var new_len = (start - end + step - 1) // step
+            var dest_ptr = UnsafePointer[T].alloc(new_len)
+            i = 0
+            while start > end:
+                dest_ptr[i] = self._data[start]
+                start -= step
+                i += 1
+            return Span[T, lifetime](unsafe_ptr=dest_ptr, len=new_len)
```

Comment:
How can we make sure the `dest_ptr` allocated memory is destroyed when the span is deleted?

Reply:
Hmmm, not sure since I don't really understand lifetimes yet. But I think building a `var items = List[T](unsafe_ptr=dest_ptr, size=new_len, capacity=new_len)` and returning `Span[T, lifetime](items^)` will work?

Either way I think this should probably return an iterator in the future (once we have them), it doesn't make much sense to allocate for this since it is a read only view on the data.